File: windows/Scanner/Rules/AESMCP011.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP011 : IRule
{
    public string Id => "AES-MCP-011";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => server.SearchableValues().Any(AiThreatPatterns.ContainsSensitiveCredentialPath))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESMCP011.swift
public struct AESMCP011: Rule {
    public let id = "AES-MCP-011"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let touchesCredentialDir = server.searchableValues.contains { value in
                AiThreatPatterns.containsSensitiveCredentialPath(value)
            }
            guard touchesCredentialDir else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}
